optuna_dashboard/ts/components/CompareStudies.tsx

```diff
@@ -0,0 +1,279 @@
+import React, { ChangeEvent, FC, useEffect, useMemo, useState } from "react"
+import { useRecoilValue } from "recoil"
+import {
+  Card,
+  CardContent,
+  FormControl,
+  Switch,
+  Typography,
+  Box,
+  useTheme,
+} from "@mui/material"
+import Chip from "@mui/material/Chip"
+import FormControlLabel from "@mui/material/FormControlLabel"
+import Divider from "@mui/material/Divider"
+import List from "@mui/material/List"
+import ListItem from "@mui/material/ListItem"
+import ListItemButton from "@mui/material/ListItemButton"
+import ListItemText from "@mui/material/ListItemText"
+import ListSubheader from "@mui/material/ListSubheader"
+import HomeIcon from "@mui/icons-material/Home"
+
+import { actionCreator } from "../action"
+import { studySummariesState, studyDetailsState } from "../state"
+import { AppDrawer } from "./AppDrawer"
+import { GraphEdfs } from "./GraphEdfs"
+import { GraphHistories } from "./GraphHistories"
+import { useHistory, useLocation } from "react-router-dom"
+
+const useQuery = (): URLSearchParams => {
+  const { search } = useLocation()
+  return useMemo(() => new URLSearchParams(search), [search])
+}
+
+const useSelectedStudies = (
+  studies: StudySummary[],
+  query: URLSearchParams
+): StudySummary[] => {
+  return useMemo(() => {
+    const selected = query.get("numbers")
+    if (selected === null) {
+      return []
+    }
+    const numbers = selected
+      .split(",")
+      .map((s) => parseInt(s))
+      .filter((n) => !isNaN(n))
+    return studies.filter(
+      (t) => numbers.findIndex((n) => n === t.study_id) !== -1
+    )
+  }, [studies, query])
+}
+
+const getStudyListLink = (numbers: number[]): string => {
+  const base = URL_PREFIX + "/compare-studies"
+  if (numbers.length > 0) {
+    return base + "?numbers=" + numbers.map((n) => n.toString()).join(",")
+  }
+  return base
+}
+
+export const CompareStudies: FC<{
+  toggleColorMode: () => void
+}> = ({ toggleColorMode }) => {
+  const theme = useTheme()
+  const query = useQuery()
+  const history = useHistory()
+
+  const action = actionCreator()
+  const studies = useRecoilValue<StudySummary[]>(studySummariesState)
+  const selected = useSelectedStudies(studies, query)
```

Comment:
Like `TrialList` component, how about selecting the first item in Study list by default?

Reply:
I specified `numbers=0` here. What do you think about this solution?
https://github.com/optuna/optuna-dashboard/pull/380/files#diff-3b4445bde53115493e67a33733abac82ab9c1e1c7f3fd447091b80971898d697R190

Reply:
I followed TrialList implementation.